File: Initial_Modulations/plotting/plot_constellation.m
function plot_constellation(sig, plotTitle, refConst)
%PLOT_CONSTELLATION  Scatter plot of complex symbols.

sig = sig(:);

scatter(real(sig), imag(sig), 8, 'filled');
hold on;

% "nargin" stands for number of argument inputs (built-in func)
if nargin >= 3 && ~isempty(refConst)
    scatter(real(refConst(:)), imag(refConst(:)), 70, 'x', 'LineWidth', 1.5);
    legend('Samples', 'Reference constellation', 'Location', 'eastoutside');
else
    legend('Samples', 'Location', 'eastoutside');
end

grid on;
axis equal;
xlabel('In-Phase');
ylabel('Quadrature');
title(plotTitle);
hold off;
end

File: Phase1_Modulations/plotting/plot_constellation.py
import matplotlib.pyplot as plt
import numpy as np


def plot_constellation(sig, plot_title, ref_const=None):
    sig = np.asarray(sig).flatten()
    plt.scatter(np.real(sig), np.imag(sig), s=8, alpha=0.5)
    if ref_const is not None:
        ref_const = np.asarray(ref_const).flatten()
        plt.scatter(
            np.real(ref_const),
            np.imag(ref_const),
            s=70,
            marker="x",
            linewidths=1.5,
            c="red",
        )
        plt.legend(["Samples", "Reference constellation"], loc="best")
    else:
        plt.legend(["Samples"], loc="best")

    plt.grid(True, linestyle="--", linewidth=0.5)
    plt.axis("equal")
    plt.xlabel("In-Phase")
    plt.ylabel("Quadrature")
    plt.title(plot_title)
